src/github.com/matrix-org/dendrite/common/transactions/transactions.go

```diff
@@ -0,0 +1,86 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transactions
+
+import (
+	"sync"
+	"time"
+
+	"github.com/matrix-org/util"
+)
+
+// DefaultCleanupPeriod represents the default time duration after which cacheCleanService runs.
+const DefaultCleanupPeriod time.Duration = 30 * time.Minute
+
+type txnsMap map[string]*util.JSONResponse
+
+// Cache represents a temporary store for response entries.
+// Entries are evicted after a certain period, defined by cleanupPeriod.
+type Cache struct {
+	sync.RWMutex
+	txnsMaps      [2]txnsMap
+	cleanupPeriod time.Duration
+}
+
+// New creates a new Cache object, starts cacheCleanService.
+// Takes cleanupPeriod (in minutes) as argument, in case of 0 DefaultCleanupPeriod is used instead.
+// Returns a reference to newly created Cache.
+func New(cleanupPeriodInMins int) *Cache {
```

Comment:
Given we have a `time.Duration` type, I'd use that. If you want to make it optional then either take a pointer and check for null or have `New()` and `NewWithPeriod(..)`

Reply:
In most scenario one would let the Default be as it is, passing nil for that case is cumbersome.
So I will avoid the nil-able pointer parameter.